Split into sub-packages to match upstream pypi and "pyg" channel packages

jamesmyatt opened the pull request:
Uses sub-package to give parity with upstream pypi and conda packages and allow install on Windows (#25).

* Adds pyg subpackage with parity for:
  * "torch_geometric" pypi package: https://github.com/pyg-team/pytorch_geometric/blob/2.6.1/pyproject.toml
  * "pyg" and "pytorch-geometric" packages in "pyg" channel: https://github.com/pyg-team/pytorch_geometric/tree/2.6.1/conda
* Adds pyg-graphgym package with parity for "torch_geometric[graphgym]" pypi package: https://github.com/pyg-team/pytorch_geometric/blob/2.6.1/pyproject.toml
* Adds pyg-modelhub package with parity for "torch_geometric[modelhub]" pypi package: https://github.com/pyg-team/pytorch_geometric/blob/2.6.1/pyproject.toml
* Adds pyg-full package with parity for "torch_geometric[full]" pypi package (except for caveats below): https://github.com/pyg-team/pytorch_geometric/blob/2.6.1/pyproject.toml
* Made "pytorch_geometric" subpackage a synonym for "pyg-full", as it was before. I'm not keen on this since it's different from the "pyg" channel package with almost the same name ("pytorch-geometric"), but it maintains parity with the current "conda-forge" package.
* Some additional import tests

Caveats:

* Will require and admin request (https://github.com/conda-forge/admin-requests?tab=readme-ov-file#add-a-package-output-to-a-feedstock) for the new outputs.
* The upper limits on some of the "extras" are removed in order to make it easier to install the package today and on all platforms (e.g. pypi torch package doesn't have a protobuf dependency, but the conda-forge one does, which conflicts with the graphgym extras; newer numba is required for Python 3.13)
* Two "full" dependencies are missing:
  * pytorch-memlab is still not available on conda
  * pgmpy is not available for windows (c.f. #25)

_Update_: In the end, it was agreed to make the following subpackages:

* Changed "pytorch_geometric" subpackage to parity with:
  * "torch_geometric" pypi package: https://github.com/pyg-team/pytorch_geometric/blob/2.6.1/pyproject.toml
  * "pyg" and "pytorch-geometric" packages in "pyg" channel: https://github.com/pyg-team/pytorch_geometric/tree/2.6.1/conda
* Add "pytorch_geometric-graphgym" package with parity for "torch_geometric[graphgym]" pypi package: https://github.com/pyg-team/pytorch_geometric/blob/2.6.1/pyproject.toml
* Add "pytorch_geometric-modelhub" package with parity for "torch_geometric[modelhub]" pypi package: https://github.com/pyg-team/pytorch_geometric/blob/2.6.1/pyproject.toml
* Remove the dependencies of the "full" optional dependencies.

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #25 , #52

jamesmyatt committed:
Split into sub-packages to match upstream

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

jamesmyatt committed:
Fix test

jamesmyatt commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.45.4, and conda-forge-pinning 2025.02.17.08.22.26

jamesmyatt committed:
Refactor recipe
Also work around https://github.com/conda/conda-build/issues/5424

jamesmyatt committed:
Simplify recipe
Based on https://github.com/conda-forge/seaborn-feedstock/blob/a0d4a037b784b3dd7b727acb772e8b6f1f266bc4/recipe/meta.yaml

jamesmyatt committed:
Use noarch: generic
like https://github.com/conda-forge/importlib_metadata-feedstock/blob/274403de0366c130e7ef545f13c433ad1b3cd233/recipe/meta.yaml#L41

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

jamesmyatt committed:
Try just repeating the python dependencies

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `test.requires` section of `<output 1` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 2` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 3` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 4` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13372508099](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13372508099). Examine the logs at this URL for more detail.</sub>

jamesmyatt committed:
Add pytorch-cpu to test dependencies

jamesmyatt committed:
Explicit about everything

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

jamesmyatt committed:
latest pytorch builds are python 3.9+

jamesmyatt committed:
Test against pytorch 2.5.1
This is the first version with builds for every platform on conda-forge

jamesmyatt committed:
Remove upper pins
To make it build

jamesmyatt committed:
pgmpy not available on windows

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [140]
 * ❌ `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13387372083](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13387372083). Examine the logs at this URL for more detail.</sub>

jamesmyatt committed:
Avoid selector

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

jamesmyatt commented:
@conda-forge/pytorch_geometric , I think this is ready to merge, except for the admin request. Please see the description and the caveats at the top. What do you think?

mikemhenry commented:
I am in favor of these changes, could you add yourself to the maintainer list?

jamesmyatt committed:
Add jamesmyatt to maintainers

jamesmyatt commented:
Admin request: https://github.com/conda-forge/admin-requests/pull/1399

jamesmyatt commented:
per https://github.com/conda-forge/admin-requests/pull/1399, I'll update the subpackages to all use "pytorch_geometric" as the name and remove the "full" version. Also fixes #52 .

jan-janssen commented:
@jamesmyatt I like the changes, please go ahead.

jamesmyatt committed:
Update following review
Fixes #52

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ pin_compatible should be used instead of pin_subpackage for `pyg` because it is not a known output of this recipe: ['pytorch_geometric', 'pytorch_geometric-graphgym', 'pytorch_geometric-modelhub', 'pytorch_geometric'].
 * ❌ pin_compatible should be used instead of pin_subpackage for `pyg` because it is not a known output of this recipe: ['pytorch_geometric', 'pytorch_geometric-graphgym', 'pytorch_geometric-modelhub', 'pytorch_geometric'].


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13465801427](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13465801427). Examine the logs at this URL for more detail.</sub>

jamesmyatt committed:
Fix subpackage pin

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

jamesmyatt commented:
@conda-forge-admin, please restart ci

mikemhenry reviewed (approved):
LGTM! Thanks for this!

mikemhenry commented:
@jamesmyatt Is it ready to merge?

jamesmyatt commented:
@mikemhenry yes I think so
